src/Dictionaries/DictionarySourceHelpers.cpp

```diff
@@ -50,4 +53,31 @@ void formatKeys(
     out->flush();
 }
 
+Context copyContextAndApplySettings(
+    const std::string & config_prefix,
+    const Context & context,
+    const Poco::Util::AbstractConfiguration & config)
+{
+    Context local_context(context);
+    if (config.has(config_prefix + ".settings"))
+    {
+        const auto prefix = config_prefix + ".settings";
+
+        Poco::Util::AbstractConfiguration::Keys config_keys;
+        config.keys(prefix, config_keys);
+
+        SettingsChanges changes;
+
+        for (const std::string & key : config_keys)
+        {
+            const auto value = config.getString(prefix + "." + key);
+            changes.emplace_back(key, value);
+        }
+
+        // local_context.checkSettingsConstraints(changes);
```

Comment:
here I don't know why if I uncomment the line my query will get segfault...

Reply:
It's because global context does not have user_id and `ContextAccess` (which contains settings constraints) is `nullptr`. However, for XML-dictionaries (unlike DDL-dictionaries) this check is needed only to report about suspicious configuration. I didn't find easy way to get settings constrains for default profile from global context, so we can just skip this check